### PR TITLE
Modified Java tests to run on Android

### PR DESCRIPTION
**Pull request by alexlyulkov**
Modified Java tests to run on Android.

To run the tests you need to:

1. Build OpenCV using Android pipeline. For example:
`cmake -DBUILD_TEST=ON -DANDROID=ON -DANDROID_ABI=arm64-v8a -DCMAKE_TOOLCHAIN_FILE=/usr/lib/android-sdk/ndk/25.1.8937393/build/cmake/android.toolchain.cmake -DANDROID_NDK=/usr/lib/android-sdk/ndk/25.1.8937393 -DANDROID_SDK=/usr/lib/android-sdk ../opencv`
`make`
2. Connect Android Phone
3. Run tests:
`cd android_tests`
`./gradlew tests_module:connectedAndroidTest`

Related CI pipeline: https://github.com/opencv/ci-gha-workflow/pull/138

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/java/test/pure_test/src/org/opencv/test/OpenCVTestCase.java [hunk 1]
```diff
@@ -106,16 +106,7 @@ public TestSkipException() {}
     protected void setUp() throws Exception {
         super.setUp();
 
-        try {
```
Comment (alexlyulkov) [inline modules/java/test/pure_test/src/org/opencv/test/OpenCVTestCase.java:109]: Removed try-cache to make the exception visible in the test report

File: modules/java/test/android_test/settings.gradle [hunk 2]
```diff
@@ -0,0 +1,6 @@
+rootProject.name = 'android_test'
+
+include ':opencv'
+project(':opencv').projectDir = new File('D:\\Android\\opencv-4.9.0-android-sdk_orig\\OpenCV-android-sdk\\sdk')
```
Comment (asmorkalov) [inline modules/java/test/android_test/settings.gradle:4]: I believe it should be cmake template. Reference to build/opencv-abdroid/sdk should be placed here.

File: modules/java/test/android_test/CMakeLists.txt [hunk 3]
```diff
@@ -1,77 +1,25 @@
...
+project(opencv_test_android)
 
-project(opencv_test_java)
+set(OPENCV_ANDROID_TEST_DIR "${OpenCV_BINARY_DIR}/android_test" CACHE INTERNAL "")
+file(REMOVE_RECURSE "${OPENCV_ANDROID_TEST_DIR}")
 
-set(OPENCV_JAVA_TEST_DIR "${OpenCV_BINARY_DIR}/android_test" CACHE INTERNAL "")
-file(REMOVE_RECURSE "${OPENCV_JAVA_TEST_DIR}")
-file(REMOVE "${OPENCV_DEPHELPER}/${the_module}_test_source_copy")
+list(APPEND MODULES_TO_TEST "calib3d" "core" "dnn" "features2d" "imgcodecs" "imgproc" "ml" "objdetect" "photo" "video" "videoio")
```
Comment (asmorkalov) [inline modules/java/test/android_test/CMakeLists.txt:6]: There is no need to list modules and iterate over them. All pre-module tests are collected by java bindings generator in build folder in `opencv-build/modules/java_bindings_generator/gen/test/org/opencv/test`. Just add it to to the test project.

File: modules/java/test/android_test/CMakeLists.txt [hunk 4]
```diff
@@ -1,77 +1,25 @@
...
+set(ANDROID_TESTS_SKIP_LIST "'*BarcodeDetectorTest*', '*QRCodeDetectorTest*'")
```
Comment (asmorkalov) [inline modules/java/test/android_test/CMakeLists.txt:7]: Skips should be implemented in the test code.

**Comment by asmorkalov** [comment]
Please remove gradlew and gradlew.bat with related files as you use `platforms/android`.

File: modules/java/test/android_test/CMakeLists.txt (hunk 3 above)
Comment (alexlyulkov) [reply modules/java/test/android_test/CMakeLists.txt]: This folder exists only when building without DANDROID

File: CMakeLists.txt [hunk 5]
```diff
@@ -1884,7 +1884,7 @@ if(BUILD_JAVA)
     status("    JNI:"         JNI_INCLUDE_DIRS    THEN "${JNI_INCLUDE_DIRS}"                                       ELSE NO)
   endif()
   status("    Java wrappers:" HAVE_opencv_java                                                            THEN "YES (${OPENCV_JAVA_SDK_BUILD_TYPE})" ELSE NO)
-  status("    Java tests:"    BUILD_TESTS AND opencv_test_java_BINARY_DIR                                 THEN YES ELSE NO)
+  status("    Java tests:"    BUILD_TESTS AND (opencv_test_java_BINARY_DIR OR ANDROID)                    THEN YES ELSE NO)
```
Comment (opencv-alalek) [inline CMakeLists.txt:1887]: It is possible to build Android native code only. Without any Java code or Java tests.

Check should be fixed.

File: modules/java/test/android_test/tests_module/build.gradle.in [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+android {
+    namespace 'org.opencv.tests'
+    compileSdkVersion @ANDROID_COMPILE_SDK_VERSION@
+    defaultConfig {
+        applicationId "org.opencv.tests"
+        minSdkVersion @ANDROID_MIN_SDK_VERSION@
+        targetSdkVersion @ANDROID_TARGET_SDK_VERSION@
+        versionCode 301
+        versionName "3.01"
```
Comment (opencv-alalek) [inline modules/java/test/android_test/tests_module/build.gradle.in:11]: > version

is it correct?

Comment (asmorkalov) [reply modules/java/test/android_test/tests_module/build.gradle.in:11]: Not sure, but it's the same everywhere. Looks like we need to substitute OpenCV version here and in samples.

Comment (asmorkalov) [reply modules/java/test/android_test/tests_module/build.gradle.in:11]: Let's do it in a separate PR.

**Review by asmorkalov — APPROVED** [review]
:+1:

File: CMakeLists.txt (hunk 5 above)
Comment (asmorkalov) [reply CMakeLists.txt]: Fixed.

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**